webapp/components/wpt-results.html

```diff
@@ -437,37 +425,45 @@ <h3>
         super.disconnectedCallback();
       }
 
-      ready() {
-        super.ready();
-        this.loadRuns().then(async runs => {
-          this.fetchResults();
-
-          // Load a diff data into this.diffRun, if needed.
-          if (this.diff && runs && runs.length === 2) {
-            this.diffRun = {
-              revision: 'diff',
-              browser_name: 'diff',
-            };
-          }
-        });
+      async ready() {
+        await super.ready();
 
         // NOTE(lukebjerring): Overriding the pathUpdated method doesn't get
-        // called, so we wrap any given onpathupdated methd here.
+        // called, so we wrap any given onpathupdated method here.
         const onpathupdated = this.onpathupdated;
         this.onpathupdated = path => {
           onpathupdated && onpathupdated(path);
           this.showHistory = false;
         };
+
+        this.loadingCount++;
+        try {
+          await this.loadRuns().then(async runs => {
+            this.fetchResults();
+
+            // Load a diff data into this.diffRun, if needed.
+            if (this.diff && runs && runs.length === 2) {
+              this.diffRun = {
+                revision: 'diff',
+                browser_name: 'diff',
+              };
+            }
+          });
+        } finally {
+          this.loadingCount--;
+        }
       }
 
       async fetchResults() {
         this.loadingCount++;
         try {
-          const response = await window.fetch(this.searchURL);
-          if (response.status !== 200) {
-            throw response;
-          }
-          const json = await response.json();
+          const json = await window.fetch(this.searchURL)
+            .then(async r => {
```

Comment:
What change in behaviour are we trying to get by switching to `.then` inside an already async function? It seems like this change could just be a change in what gets thrown  when `response.status !== 200`.

Reply:
Consistency with the exact same chunk of code in `interop`, which will make for a cleaner PR when the 2 get consolidated.